h2o-extensions/target-encoder/src/jmh/java/hex/NewRGSBench.java

```diff
@@ -0,0 +1,100 @@
+package hex;
+
+import ai.h2o.targetencoding.TargetEncoderModel;
+import hex.grid.GridSearch;
+import hex.grid.HyperSpaceSearchCriteria;
+import hex.grid.HyperSpaceWalker;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.profile.StackProfiler;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * PojoUtils benchmark
+ */
+@State(Scope.Thread)
+@Fork(value = 1, jvmArgsAppend = "-Xmx12g")
+@Warmup(iterations = 2)
+@Measurement(iterations = 3)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class NewRGSBench {
+
+
+  @Benchmark
+  public void traverseWholeGridNew() {
+    HashMap<String, Object[]> hpGrid = new HashMap<>();
```

Comment:
this seems to be a copy of code that is in the MaterializedRandomWalker

Reply:
@michalkurka yes, hence the comment here https://github.com/h2oai/h2o-3/pull/4021/files#diff-73353bb7545487cc487e075203b698feR31 . Firstly I was benchmarking the core logic for traversing the grid then I've integrated it into MaterialisedRandomWalker and did separate benchmarking of the logic+walker_wrapper. In the end we can remove redundant  benchmarks